### PR TITLE
Add condition variable to set_trace

### PR DESCRIPTION
**Pull request by alexandrebarbaruiva**
Add condition variable to set_trace to show console only when ``cond=True``.

This way, instead of doing
```
for abc in range(1, 100):
    if abc == 13:
        import ipdb; ipdb.set_trace()
```
You do this
```
for abc in range(1, 100):
    import ipdb; ipdb.set_trace(cond=(abc==13))
```

### CODE REVIEW

File: ipdb/__main__.py [hunk 1]
```diff
@@ -64,17 +64,18 @@ def wrap_sys_excepthook():
...
-        context = os.environ.get(
-            "IPDB_CONTEXT_SIZE", get_context_from_config()
-        )
-    if frame is None:
-        frame = sys._getframe().f_back
-    p = _init_pdb(context).set_trace(frame)
-    if p and hasattr(p, 'shell'):
-        p.shell.restore_sys_module_state()
+def set_trace(frame=None, context=None, cond=True):
+    if cond:
```
Comment (gotcha) [inline ipdb/__main__.py:68]: I'd rather go:

    if not cond:
        return

and avoid indentation.

**Comment by gotcha** [comment]
Great ideas. Both the condition and the documentation improvement.

**Comment by gotcha** [comment]
Before merging, I'd like to see a changelog entry... and some doc for the new arg to `set_trace`

**Comment by alexandrebarbaruiva** [comment]
Changed code based on review and added documentation and a changelog entry!

**Comment by gotcha** [comment]
Released in https://pypi.org/project/ipdb/0.13.3/ Thanks !

**Pull request merged**